### PR TITLE
Apply code reviews

### PR DESCRIPTION
**Pull request by raminqaf**
I have used the class `java.net.ServerSocket` to get a random free port every time the test is invoked. According to the JavaDocs of SeverSocket class:
```text
Creates a server socket, bound to the specified port. 
A port number of 0 means that the port number is automatically allocated, typically from an ephemeral port range. 
This port number can then be retrieved by calling getLocalPort.
```
I also fixed some small issues and reviews. There are some points that I like to talk about with you before changing them.

### CODE REVIEW

File: flink-connector-influxdb2/src/test/java/org/apache/flink/streaming/connectors/InfluxDBSourceIntegrationTestCase.java [hunk 1]
```diff
@@ -67,59 +71,73 @@
...
-                    .setPort(PORT)
-                    .setDeserializer(new InfluxDBTestDeserializer())
-                    .build();
 
     @BeforeAll
-    static void setUp() {
+    void setUp() {
         CollectSink.VALUES.clear();
+        try {
+            this.port = new ServerSocket(0).getLocalPort();
```
Comment (1p4pk) [inline flink-connector-influxdb2/src/test/java/org/apache/flink/streaming/connectors/InfluxDBSourceIntegrationTestCase.java]: This is still using the same port again.
image

Comment (1p4pk) [reply flink-connector-influxdb2/src/test/java/org/apache/flink/streaming/connectors/InfluxDBSourceIntegrationTestCase.java]: Does not fix the problem

**Review by 1p4pk — CHANGES_REQUESTED** [review]

File: flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/common/InfluxParser.java [hunk 2]
```diff
@@ -134,7 +132,8 @@ private Number parseTimestamp(@Nullable final TimestampContext timestamp) {
         if (strTimestamp.length() < 7) {
             return 0L;
         } else {
-            return Long.valueOf(strTimestamp.substring(0, strTimestamp.length() - 6));
+            //            return Long.valueOf(strTimestamp.substring(0, strTimestamp.length() - 6));
+            return Long.valueOf(strTimestamp);
```
Comment (raminqaf) [inline flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/common/InfluxParser.java]: @1p4pk We should talk about timestamp parsing. I don't see the point to do the substring.

File: flink-connector-influxdb2/README.md [hunk 3]
```diff
@@ -76,16 +76,17 @@ class TestDeserializer implements InfluxDBDataPointDeserializer<Long> {
...
+| ------------- |:-------------:|
 |    Float      | ✅            |
 |    Integer    | ✅            |
 |    UInteger   | ❌            |
 |    String     | ✅            |
 |    Boolean    | ✅            |
 
 See InfluxDB field set value [data type](https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/#field-set).
- 
+The parsing limitation is related to the Apache Druid project. For more information see this [issue](https://github.com/apache/druid/issues/10993)
```
Comment (1p4pk) [inline flink-connector-influxdb2/README.md:88]: Is this still a limitation? Thought this was solved?

File: flink-connector-influxdb2/pom.xml [hunk 4]
```diff
@@ -38,7 +38,6 @@ under the License.
   <properties>
     <influxdbClient.version>2.0.0</influxdbClient.version>
     <flink.new.version>1.12.2</flink.new.version>
```
Comment (1p4pk) [inline flink-connector-influxdb2/pom.xml]: We should remove flink and scala version here and use the one from the main pom. Maybe merge the current master into our master and then the master into our dev

File: flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/common/DataPoint.java [hunk 5]
```diff
@@ -72,9 +84,10 @@ public void addField(final String field, final Object value) {
      * @param field Key of field.
      * @return value Value for the field key.
      */
-    public Object getField(final String field) {
+    @SuppressWarnings("unchecked")
+    public <T> T getField(final String field) {
```
Comment (1p4pk) [inline flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/common/DataPoint.java:88]: Why not remove Object?

File: flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/common/DataPoint.java [hunk 6]
```diff
@@ -98,4 +111,47 @@ public String getTag(final String key) {
...
+     * conflicts favor the new field set.
+     *
+     * @see <a
+     *     href="https://docs.influxdata.com/influxdb/cloud/reference/syntax/line-protocol/#duplicate-points">
+     *     Duplicate points </a>
+     * @param obj: Object to compare to
+     * @return Either the object is equal to the data point or not
+     */
+    @Override
+    public boolean equals(final Object obj) {
```
Comment (1p4pk) [inline flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/common/DataPoint.java:132]: Where do you use this?

File: flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/common/InfluxParser.java (hunk 2 above)
Comment (1p4pk) [reply flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/common/InfluxParser.java]: I thought we had removed that already. Or was it only in the benchmark branch?

Comment (1p4pk) [reply flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/common/InfluxParser.java]: Yes, we only removed that in the benchmark branch. Just checked

Comment (1p4pk) [reply flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/common/InfluxParser.java]: Then remove the line before
```suggestion
            return Long.valueOf(strTimestamp);
```

File: flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/writer/InfluxDBWriter.java [hunk 7]
```diff
@@ -73,26 +77,22 @@ public InfluxDBWriter(
...
+            LOG.debug("Buffer size reached preparing to write the elements.");
+            this.writeCurrentElements();
+            this.elements.clear();
+        } else {
+            LOG.trace("Adding elements to buffer. Buffer size: {}", this.elements.size());
+            this.elements.add(this.schemaSerializer.serialize(in, context));
+            if (context.timestamp() != null) {
+                this.lastTimestamp = Math.max(this.lastTimestamp, context.timestamp());
             }
-        } catch (final Exception e) {
```
Comment (1p4pk) [inline flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/sink/writer/InfluxDBWriter.java:89]: What is the reason that you removed this exception?

File: flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/source/http/HealthCheckHandler.java [hunk 8]
```diff
@@ -20,9 +20,16 @@
 import com.sun.net.httpserver.HttpExchange;
 import java.io.IOException;
 
+/**
+ * Handles incoming health check requests from /health path. If the server is running a response
+ * code 200 is sent
+ */
 public final class HealthCheckHandler extends Handler {
+
+    public static final int HTTP_STATUS_OK = 200;
```
Comment (1p4pk) [inline flink-connector-influxdb2/src/main/java/org/apache/flink/streaming/connectors/influxdb/source/http/HealthCheckHandler.java]: Why our own?

File: flink-connector-influxdb2/src/test/java/org/apache/flink/streaming/connectors/influxdb/InfluxDBSourceIntegrationTestCase.java [hunk 9]
```diff
@@ -67,59 +68,76 @@
...
+                this.writeToInfluxDB("test longValue=1i 1\ntest longValue=2i 2");
 
         assertEquals(writeResponseCode, HttpURLConnection.HTTP_NO_CONTENT);
 
         jobClient.cancel();
 
         final Collection<Long> results = new ArrayList<>();
         results.add(2L);
+        results.add(3L);
+        Thread.sleep(500);
```
Comment (1p4pk) [inline flink-connector-influxdb2/src/test/java/org/apache/flink/streaming/connectors/influxdb/InfluxDBSourceIntegrationTestCase.java]: Not good, hardcode a thread sleep here is a mistake by design. Should think about something else

File: flink-connector-influxdb2/src/test/java/org/apache/flink/streaming/connectors/influxdb/common/InfluxParserTest.java [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+    void shouldParseNotDuplicatedLineToDataPoint() throws ParseException {
+        final String lineProtocol =
+                "test,testTagKey=testTagValue testFieldKey=\"testFieldValue\" 1556813561098000000";
+        final DataPoint dataPoint = InfluxParser.parseToDataPoint(lineProtocol);
+
+        final String lineProtocol2 =
+                "test,testTagKey=diff testFieldKey=\"testFieldValue\" 1556813561098000000";
+        final DataPoint dataPoint2 = InfluxParser.parseToDataPoint(lineProtocol2);
+
+        assertNotEquals(dataPoint, dataPoint2);
```
Comment (1p4pk) [inline flink-connector-influxdb2/src/test/java/org/apache/flink/streaming/connectors/influxdb/common/InfluxParserTest.java]: Also add assertequals to the correct data points

File: flink-connector-influxdb2/src/test/java/org/apache/flink/streaming/connectors/influxdb/InfluxDBSourceIntegrationTestCase.java (hunk 9 above)
Comment (1p4pk) [reply flink-connector-influxdb2/src/test/java/org/apache/flink/streaming/connectors/influxdb/InfluxDBSourceIntegrationTestCase.java]:
```suggestion
```

**Review by 1p4pk — APPROVED** [review]

**Pull request merged**